8359181: Error messages generated by configure --help after 8301197

sendaoYan opened the pull request:
Hi all,

The macro `HELP_PRINT_ADDITIONAL_HELP_AND_EXIT` is only called directly from make/autoconf/configure. In JDK-8301197,  most uses of `printf` was changed to `echo`, but this explicit call wasn't updated to also set ` ECHO=echo`, this PR add ` ECHO=echo` when call this macro.

Change has been verified locally, almost no risk.

<!-- Anything below this marker will be automatically updated, please do not edit manually! -->
---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8359181](https://bugs.openjdk.org/browse/JDK-8359181): Error messages generated by configure --help after 8301197 (**Bug** - P4)


### Reviewers
 * [Erik Joelsson](https://openjdk.org/census#erikj) (@erikj79 - **Reviewer**)
 * [Magnus Ihse Bursie](https://openjdk.org/census#ihse) (@magicus - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/25740/head:pull/25740` \
`$ git checkout pull/25740`

Update a local copy of the PR: \
`$ git checkout pull/25740` \
`$ git pull https://git.openjdk.org/jdk.git pull/25740/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 25740`

View PR using the GUI difftool: \
`$ git pr show -t 25740`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/25740.diff">https://git.openjdk.org/jdk/pull/25740.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/25740#issuecomment-2961133497)
</details>

erikj79 reviewed (commented):
This does not seem like the correct solution. This macro is only called directly from `make/autoconf/configure`, which is our wrapper script around the autoconf generated scripts. That wrapper expects this macro to be using `printf` so it sets `PRINTF=printf` on the command line:
https://github.com/openjdk/jdk/blob/9586817cea3f1cad8a49d43e9106e25dafa04765/make/autoconf/configure#L369
In JDK-8301197, we changed most uses of `printf` to `echo`, but this explicit call wasn't updated to also set `ECHO=echo`. Doing that fixes the issue.

erikj79 reviewed (approved):
I think this looks good, but please give Magnus a chance to review as his opinion may differ and this is more of his area than mine.

sendaoYan commented:
> In JDK-8301197, we changed most uses of `printf` to `echo`, but this explicit call wasn't updated to also set `ECHO=echo`. Doing that fixes the issue.

@erikj79 Thanks your correction, I have revert the original commit and add ` ECHO=echo` when call the mcaro, the PR description also updated.

sendaoYan commented:
Thanks @magicus @erikj79 for the reviews and suggestions.

/integrate
